frontend/src/routes/manage/Realm/DangerZone.tsx

```diff
@@ -209,25 +204,21 @@ const RemoveRealm: React.FC<InnerProps> = ({ realm }) => {
             {t("manage.realm.danger-zone.delete.warning")}
         </p>
         <div css={{ marginTop: 32, textAlign: "center" }}>
-            <Button kind="danger" onClick={() => setModalActive(true)}>
+            <Button kind="danger" onClick={() => modalRef.current?.open()}>
```

Comment:
Regarding the first point: No. Optional chaining short-circuits:

```javascript
({}).foo?.bar()
```

is just `undefined`.

`!` is currently not allowed by our ESLint rules. Because of that I have been going with things like

```javascript
const theRealThing = theThing ?? bug("the thing is not real");
```

elsewhere. I could do something like this here as well. The advantage is that you are forced to make the failure explicit. Most of the time if you would use `!` wrongly you would get an error anyway, like in this case, because you try to access stuff on `undefined`, but in general one can imagine cases where constructs with `bug` fail loudly while `!` would just lead to weird behavior.

What do you think? :thinking:

Reply:
Yeah right, `!` is not a runtime check at all right? So ye, `?? bug` is better in that regard. In this case, I don't care. The current version is fine too.

Reply:
Exactly, `!` is just stripped out during compilation. It's just to appease the type checker.

The problem with the `bug` approach in this particular case is that you have to spell it out every time. You can't memoize it on the top level of a component, precisely because it's a ref. What do you think about a helper function, maybe in `err`, a la

```javascript
const currentRef = ref => ref.current ?? bug("unbound ref");
```

?

Reply:
Yeah why not. Sounds like a use case that happens often enough. "Ref should be bound in handler" or sth. Ye.

Reply:
Ok, it's done in e072ca3. I also transitioned some of your code to it. Just the obvious cases, though. You might want to double check me. :shrug: